api/src/infrastructure/kubernetes/infrastructure.rs

```diff
@@ -500,20 +494,26 @@ impl KubernetesInfrastructure {
         &self,
         app_name: &str,
         service: &'a DeployableService,
-    ) -> Result<HashMap<&'a String, PersistentVolumeClaim>, KubernetesInfrastructureError> {
-        let client = self.client().await?;
+    ) -> Option<HashMap<&'a String, PersistentVolumeClaim>> {
+        let client = match self.client().await.ok() {
+            Some(val) => val,
+            None => {
+                return None;
+            }
+        };
 
-        let Runtime::Kubernetes(k8s_config) = self.config.runtime_config() else {return Ok(HashMap::new())};
+        let Runtime::Kubernetes(k8s_config) = self.config.runtime_config() else {return None};
 
         let storage_size = k8s_config.storage_config().storage_size();
         let storage_class = match k8s_config.storage_config().storage_class() {
             Some(sc) => Some(sc.into()),
-            None => self.fetch_default_storage_class().await,
+            None => self
+                .fetch_default_storage_class()
+                .await
+                .map_or_else(|| None, |sc| sc.metadata.name),
         };
 
-        if storage_class.is_none() {
-            return Ok(HashMap::new());
-        }
+        storage_class.as_ref()?;
```

Comment:
That is kind of an odd pattern here. I would like to see a change where `storage_class` won't be an Option in the first place rather than having a single line using just the `?` operator.

Reply:
This has been avoided and now storage class is a `&str`